Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -344,16 +344,39 @@ TextureCache::TCacheEntryBase* TextureCache::Load(const u32 stage)
 		//       already is an efb copy at this source. This makes those textures less broken when using efb to texture.
 		//       Examples are the mini map in Twilight Princess and objects on the targetting computer in Rogue Squadron 2(RS2).
 		// TODO: Convert those textures using the right palette, so they display correctly
-		auto iter = textures.find(texID);
-		if (iter == textures.end() || !iter->second->IsEfbCopy())
-			texID ^= ((u32)tlut_hash) ^(u32)(tlut_hash >> 32);
 	}
 
 	// GPUs don't like when the specified mipmap count would require more than one 1x1-sized LOD in the mipmap chain
 	// e.g. 64x64 with 7 LODs would have the mipmap chain 64x64,32x32,16x16,8x8,4x4,2x2,1x1,0x0, so we limit the mipmap count to 6 there
 	tex_levels = std::min<u32>(IntLog2(std::max(width, height)) + 1, tex_levels);
 
-	TCacheEntryBase*& entry = textures[texID];
+	TCacheEntryBase* entry = nullptr;
+
+	// Allow multiple cache entries for textures at the same address. This is a trick to make the Castlevania 3 faster, which
+	// uses the same address for different textures. Do not do this for paletted textures, because there's a similar trick used
+	// for them already. And do not do this for efb copies, because those would break with this. To prevent a regression caused
+	// by constantly creating new textures instead of overwriting old ones, only do this for textures that are loaded at the
+	// same frame.
+	TexCache::iterator iter = textures.lower_bound(texID);
+	if (iter != textures.end())
+	{
+		if (iter->second->IsEfbCopy() || (iter->second->frameCount != FRAMECOUNT_INVALID && textures.count(texID) == 1))
+		{
```

Comment:
Thanks, now i understand. Is there a faster/better way than:
if (iter != textures.end() && iter != textures.upper_bound(texID))
?
Hmm, i could do this:
if (textures.count(texID) != 0)
{
TexCache::iterator iter = textures.lower_bound(texID);
if (iter->second->IsEfbCopy() [...]

I think everytime i use lower_bound, upper_bound or count, the whole map is searched, so i should try to avoid calling those more ofthen than neccessary.